src/communication/stmt.rs

```diff
@@ -0,0 +1,153 @@
+        self.body
+            .replace_communication_call(modules, module, class, method);
+        if let Some(else_body) = self.else_body.as_mut() {
+            else_body.replace_communication_call(modules, module, class, method);
+        }
+        None
+    }
+}
+
+impl CommunicationReplacer for ForStmt {
+    fn replace_communication_call(
+        &mut self,
+        modules: &Vec<ModuleComponent>,
+        module: &ModuleComponent,
+        class: Option<&ClassOrInterfaceComponent>,
+        method: &MethodComponent,
+    ) -> Option<Node> {
+        self.body
+            .replace_communication_call(modules, module, class, method)
+    }
+}
+impl CommunicationReplacer for ForRangeStmt {
+    fn replace_communication_call(
+        &mut self,
+        modules: &Vec<ModuleComponent>,
+        module: &ModuleComponent,
+        class: Option<&ClassOrInterfaceComponent>,
+        method: &MethodComponent,
+    ) -> Option<Node> {
+        self.body
+            .replace_communication_call(modules, module, class, method)
+    }
+}
+
+impl CommunicationReplacer for WhileStmt {
+    fn replace_communication_call(
+        &mut self,
+        modules: &Vec<ModuleComponent>,
+        module: &ModuleComponent,
+        class: Option<&ClassOrInterfaceComponent>,
+        method: &MethodComponent,
+    ) -> Option<Node> {
+        self.body
+            .replace_communication_call(modules, module, class, method)
+    }
+}
+impl CommunicationReplacer for DoWhileStmt {
+    fn replace_communication_call(
+        &mut self,
+        modules: &Vec<ModuleComponent>,
+        module: &ModuleComponent,
+        class: Option<&ClassOrInterfaceComponent>,
+        method: &MethodComponent,
+    ) -> Option<Node> {
+        self.body
+            .replace_communication_call(modules, module, class, method)
+    }
+}
+
+impl CommunicationReplacer for SwitchStmt {
+    fn replace_communication_call(
+        &mut self,
+        modules: &Vec<ModuleComponent>,
+        module: &ModuleComponent,
+        class: Option<&ClassOrInterfaceComponent>,
+        method: &MethodComponent,
+    ) -> Option<Node> {
+        for case in self.cases.iter_mut() {
+            case.replace_communication_call(modules, module, class, method);
+        }
+        None
+    }
+}
+
+impl CommunicationReplacer for CaseStmt {
+    fn replace_communication_call(
+        &mut self,
+        modules: &Vec<ModuleComponent>,
+        module: &ModuleComponent,
+        class: Option<&ClassOrInterfaceComponent>,
+        method: &MethodComponent,
+    ) -> Option<Node> {
+        self.body
+            .replace_communication_call(modules, module, class, method)
+    }
+}
+
+impl CommunicationReplacer for TryCatchStmt {
+    fn replace_communication_call(
+        &mut self,
+        modules: &Vec<ModuleComponent>,
+        module: &ModuleComponent,
+        class: Option<&ClassOrInterfaceComponent>,
+        method: &MethodComponent,
+    ) -> Option<Node> {
+        self.try_body
```

Comment:
What about catch bodies?

We'll also want to add `finally`, when my branch gets merged in (I'd added that as a 100% optional construct).

Reply:
not sure how commonly you'd be making API calls inside of a catch, but could be added pretty easily since it's just another block.

Good call with finally statements

Reply:
Added catch bodies to this